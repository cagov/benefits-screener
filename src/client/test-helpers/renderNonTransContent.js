import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
