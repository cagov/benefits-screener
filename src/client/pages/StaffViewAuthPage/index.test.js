import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import AUTH_STRINGS from "../../../data/authStrings";
import Component from "./index";

describe("<StaffViewAuthPage />", () => {
  it("retro certs auth page", async () => {
    const wrapper = renderNonTransContent(Component);

    expect(wrapper).toMatchSnapshot();
  });

  it("retro certs auth page with user not found error", async () => {
    const wrapper = renderNonTransContent(Component, {
      userData: { status: AUTH_STRINGS.statusCode.userNotFound },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
