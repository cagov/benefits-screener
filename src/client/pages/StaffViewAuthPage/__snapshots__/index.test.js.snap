// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffViewAuthPage /> retro certs auth page 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <h2
        className="h3 font-weight-bold mb-3"
      >
        Staff View
      </h2>
      <p>
        Enter claimant information below.
      </p>
      <p>
        * Indicates required text
      </p>
      <Form
        inline={false}
        noValidate={true}
        onSubmit={[Function]}
        validated={false}
      >
        <Row
          noGutters={false}
        >
          <FormGroup
            className="col-md-6"
            controlId="formLastName"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              * Last Name
            </FormLabel>
            <FormControl
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <div
          className="mb-n2 mt-4"
        >
          * Date of Birth
        </div>
        <small
          className="text-muted"
        >
          For example, 4 28 1990
        </small>
        <Row
          noGutters={false}
        >
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobMonth"
            md={2}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Month
            </FormLabel>
            <FormControl
              maxLength={2}
              onChange={[Function]}
              pattern="(^0[1-9])|(^1[0-2])|(^[1-9]$)"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobDay"
            md={2}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Day
            </FormLabel>
            <FormControl
              maxLength={2}
              onChange={[Function]}
              pattern="(^0[1-9])|(^1[0-9])|(^2[0-9])|(^3[0-1])|(^[1-9]$)"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobYear"
            md={3}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Year
            </FormLabel>
            <FormControl
              maxLength={4}
              onChange={[Function]}
              pattern="[12][890]\\\\d\\\\d"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <Row
          className="mt-4"
          noGutters={false}
        >
          <FormGroup
            className="col-md-6"
            controlId="formSsn"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              * Social Security Number
            </FormLabel>
            <FormText
              className="ssn-hint"
              muted={true}
            >
              Enter your Social Security number with or without dashes.
            </FormText>
            <FormControl
              aria-describedby="ssn-sr-desc"
              onChange={[Function]}
              pattern="\\\\d\\\\d\\\\d[-]?\\\\d\\\\d[-]?\\\\d\\\\d\\\\d\\\\d"
              required={true}
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <Button
          active={false}
          className="mt-4"
          disabled={false}
          type="submit"
          variant="secondary"
        >
          Search
        </Button>
      </Form>
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;

exports[`<StaffViewAuthPage /> retro certs auth page with user not found error 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <h2
        className="h3 font-weight-bold mb-3"
      >
        Staff View
      </h2>
      <p>
        Enter claimant information below.
      </p>
      <p>
        * Indicates required text
      </p>
      <Form
        inline={false}
        noValidate={true}
        onSubmit={[Function]}
        validated={false}
      >
        <Row
          noGutters={false}
        >
          <FormGroup
            className="col-md-6"
            controlId="formLastName"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              * Last Name
            </FormLabel>
            <FormControl
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <div
          className="mb-n2 mt-4"
        >
          * Date of Birth
        </div>
        <small
          className="text-muted"
        >
          For example, 4 28 1990
        </small>
        <Row
          noGutters={false}
        >
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobMonth"
            md={2}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Month
            </FormLabel>
            <FormControl
              maxLength={2}
              onChange={[Function]}
              pattern="(^0[1-9])|(^1[0-2])|(^[1-9]$)"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobDay"
            md={2}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Day
            </FormLabel>
            <FormControl
              maxLength={2}
              onChange={[Function]}
              pattern="(^0[1-9])|(^1[0-9])|(^2[0-9])|(^3[0-1])|(^[1-9]$)"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Col",
                "render": [Function],
              }
            }
            controlId="formDobYear"
            md={3}
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Year
            </FormLabel>
            <FormControl
              maxLength={4}
              onChange={[Function]}
              pattern="[12][890]\\\\d\\\\d"
              required={true}
              type="text"
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <Row
          className="mt-4"
          noGutters={false}
        >
          <FormGroup
            className="col-md-6"
            controlId="formSsn"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              * Social Security Number
            </FormLabel>
            <FormText
              className="ssn-hint"
              muted={true}
            >
              Enter your Social Security number with or without dashes.
            </FormText>
            <FormControl
              aria-describedby="ssn-sr-desc"
              onChange={[Function]}
              pattern="\\\\d\\\\d\\\\d[-]?\\\\d\\\\d[-]?\\\\d\\\\d\\\\d\\\\d"
              required={true}
              value=""
            />
            <Feedback
              type="invalid"
            >
              This field is required.
            </Feedback>
          </FormGroup>
        </Row>
        <Row
          className="mt-3"
          noGutters={false}
        >
          <div
            className="col-md-12"
          >
            <Alert
              closeLabel="Close alert"
              show={true}
              transition={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "appear": false,
                    "in": false,
                    "mountOnEnter": false,
                    "timeout": 300,
                    "unmountOnExit": false,
                  },
                  "displayName": "Fade",
                  "render": [Function],
                }
              }
              variant="danger"
            >
              This claimant does not need to retroactively certify.
            </Alert>
          </div>
        </Row>
        <Button
          active={false}
          className="mt-4"
          disabled={false}
          type="submit"
          variant="secondary"
        >
          Search
        </Button>
      </Form>
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;
